codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/AsyncClientClass.java

```diff
@@ -144,9 +146,19 @@ public TypeSpec poetSpec() {
             classBuilder.addField(FieldSpec.builder(ClassName.get(ScheduledExecutorService.class), "executorService")
                                                    .addModifiers(PRIVATE, FINAL)
                                                    .build());
+            hasScheduledExecutor = true;
             classBuilder.addMethod(waiterImplMethod());
         }
 
+        if (model.getCustomizationConfig().getBatchManagerMethod() != null) {
+            classBuilder.addMethod(batchMangerMethod());
+            if (!hasScheduledExecutor) {
+                classBuilder.addField(FieldSpec.builder(ClassName.get(ScheduledExecutorService.class), "executorService")
+                                               .addModifiers(PRIVATE, FINAL)
+                                               .build());
+            }
+        }
```

Comment:
Can we extract the logic to a separate method, say `addScheduledExecutorIfNeeded` and set/check `hasScheduledExecutor` there? It's easier to manage that way

Reply:
Extracted to a `addBatchManagerMethodAndScheduledExecutorIfNeeded` method. hasScheduledExecutor is passed as method parameter.